projects/aca-playwright-shared/src/page-objects/components/sidenav.component.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { BaseComponent } from './base.component';
+import { Locator, Page } from '@playwright/test';
+
+export class SidenavComponent extends BaseComponent {
+  private static rootElement = 'app-sidenav';
+
+  private personalFiles = this.getChild(`[data-automation-id='app.navbar.personalFiles']`);
+  private fileLibraries = this.getChild(`[data-automation-id='app.navbar.libraries.menu']`);
+  private myLibraries = this.getChild(`[data-automation-id='app.navbar.libraries.files']`);
+  private favoriteLibraries = this.getChild(`[data-automation-id='app.navbar.libraries.favorite']`);
+  private shared = this.getChild(`[data-automation-id='app.navbar.shared']`);
+  private recentFiles = this.getChild(`[data-automation-id='app.navbar.recentFiles']`);
+  private favorites = this.getChild(`[data-automation-id='app.navbar.favorites']`);
+  private trash = this.getChild(`[data-automation-id='app.navbar.trashcan']`);
+  private sidenavToggle = this.getChild(`.sidenav-header-title-logo`);
+  private sidenavExpand = this.page.getByTitle('Expand navigation menu');
+  public expandedSidenav = this.page.locator(`[data-automation-id='expanded']`);
+
+  constructor(page: Page) {
+    super(page, SidenavComponent.rootElement);
+  }
+
+  async isActive(name: string): Promise<boolean> {
+    const cssClass = await this.getLinkLabel(name).getAttribute('class');
+    return cssClass.includes('action-button--active');
+  }
+
+  async openPanel(name: string): Promise<void> {
+    await this.getLinkLabel(name).click();
+  }
+
+  private getLinkLabel(name: string): Locator {
```

Comment:
you have introduced `SIDEBAR_LABELS` so you can use it here too

Reply:
used SIDEBAR_LABELS in case statement.